src/context.rs

```diff
@@ -67,12 +67,14 @@ impl Context {
     /// The output is stdout.
     /// The returned line has the newline removed.
     /// Before returning, will revert all changes to the history buffers.
-    pub fn read_line<P: Into<String>>(
+    pub fn read_line<P: Into<String>, F: 'static>(
         &mut self,
         prompt: P,
+        f: F,
         mut handler: &mut EventHandler<RawTerminal<Stdout>>,
-    ) -> io::Result<String> {
-        self.read_line_with_init_buffer(prompt, handler, Buffer::new())
+    ) -> io::Result<String> 
+    where for<'r> F: (Fn(&'r str,) -> String) {
```

Comment:
Could can probably merge this together with `F: 'static` to get `where for<'r> F: Fn(&r str) -> String + 'static`. Not sure though.